### PR TITLE
Add observers to `RDFSource`

### PR DESCRIPTION
**Pull request by no-reply**
RDFSources now notify observers of changes.

This is an alternate implementation for #258.

### CODE REVIEW

**Comment by no-reply** [comment]
@mbarnett, @jcoyne: I wonder if this will suit as a replacement for the callbacks in #258? I prefer the observer pattern here, since we can easily have multiple observers and it avoids the marshalling issues worked around in the previous PR.

File: lib/active_triples/rdf_source.rb [hunk 1]
```diff
@@ -90,6 +91,8 @@ def type_registry
     # @see RDF::Graph
     # @todo move this logic out to a Builder?
     def initialize(*args, &block)
+      @observers   = Set.new
```
Comment (jcoyne) [inline lib/active_triples/rdf_source.rb]: I wonder if this should be a different object.  e.g. `RdfSource#observers.notify`

Comment (no-reply) [inline lib/active_triples/rdf_source.rb]: I was trying to keep the interface pretty close to `Observable` http://ruby-doc.org/stdlib-2.0.0/libdoc/observer/rdoc/Observable.html, but maybe that's not important?

Comment (jcoyne) [inline lib/active_triples/rdf_source.rb]: 👌 WFM.

**Pull request merged**